.ecp-gitlab-ci.yml

```diff
@@ -0,0 +1,110 @@
+# see https://docs.gitlab.com/ce/ci/yaml/README.html for all available options
+variables:
+  CI_DEBUG_SERVICES: trace
+  SCHEDULER_PARAMETERS: "-P STF002 -J ginsburgtest -W 0:03 -nnodes 2"
+
+stages:
+  - buildKokkos
+  - build
+#  - test
+
+.BuildKokkos:
+  stage: buildKokkos
+  script:
+    - CI_PROJECT_DIR=${PWD}
+    - module load gcc
+    - j="$(grep -c processor /proc/cpuinfo 2>/dev/null)" || j=0; ((j++))
+    - for i in ${BACKENDS}; do KOKKOS_OPTS+=( --with-${i,,[A-Z]} ); done
+    - git clone --depth=1 https://github.com/kokkos/kokkos.git &&
+      pushd kokkos &&
+      mkdir build &&
+      pushd build &&
+      ../generate_makefile.bash --prefix=${CI_PROJECT_DIR}/kokkos.install ${KOKKOS_OPTS[@]} &&
+      make -j${j} &&
+      make install &&
+      popd &&
+      popd
```

Comment:
Why did you chose to use `pushd` and `popd` rather than `cd`?  You don't seem to take any advantage of being able to return to the paths in the directory stack.  Or am I missing something?

Reply:
Just my personal preference, plus I like that it shows what directory I am in.

Reply:
https://unix.stackexchange.com/questions/272965/pushd-popd-vs-cd-cd-in-bash-and-zsh